File: member_crud-jpa/app/src/main/java/sample/member/app/web/signup/SignupRestController.java
package sample.member.app.web.signup;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import sample.member.core.transfer.EmptyResponseWrapper;

import javax.validation.Valid;

@RestController
@RequestMapping("/signup/rest")
public class SignupRestController {

	@Autowired
	private SignupService signupService;

	@RequestMapping(value = "/checkUsername", method = RequestMethod.GET)
	@ResponseBody
	public ResponseEntity checkUsername(@RequestParam String username) {
		if (!signupService.isExistsUsername(username)) {
			return ResponseEntity.ok().build();
		} else {
			return ResponseEntity.notFound().build();
		}
	}

	@RequestMapping(value = "/checkEmail", method = RequestMethod.GET)
	@ResponseBody
	public ResponseEntity checkEmail(@RequestParam String email) {
		if (!signupService.isExistsEmail(email)) {
			return ResponseEntity.ok().build();
		} else {
			return ResponseEntity.notFound().build();
		}
	}

	@RequestMapping(value = "", method = RequestMethod.POST)
	@ResponseBody
	public EmptyResponseWrapper signupAction(@RequestBody @Valid SignupRequestDto requestDto) {
		signupService.signup(requestDto);
		return EmptyResponseWrapper.create().success(true);
	}

}

File: 2017-multi-jpa-kotlin/app/src/main/kotlin/sample/member/app/web/signup/SignupRestController.kt
package sample.member.app.web.signup

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import sample.member.core.transfer.EmptyResponseWrapper

import javax.validation.Valid

@RestController
@RequestMapping("/signup/rest")
class SignupRestController {

	@Autowired
	private lateinit var signupService: SignupService

	@RequestMapping(value = ["/checkUsername"], method = [RequestMethod.GET])
	@ResponseBody
	fun checkUsername(@RequestParam username: String): ResponseEntity<*> =
			if (!signupService.isExistsUsername(username)) {
				ResponseEntity.ok().build<Any>()
			} else {
				ResponseEntity.notFound().build<Any>()
			}

	@RequestMapping(value = ["/checkEmail"], method = [RequestMethod.GET])
	@ResponseBody
	fun checkEmail(@RequestParam email: String): ResponseEntity<*> =
			if (!signupService.isExistsEmail(email)) {
				ResponseEntity.ok().build<Any>()
			} else {
				ResponseEntity.notFound().build<Any>()
			}

	@RequestMapping(value = [""], method = [RequestMethod.POST])
	@ResponseBody
	fun signupAction(@RequestBody @Valid requestDto: SignupRequestDto): EmptyResponseWrapper {
		signupService.signup(requestDto)
		return EmptyResponseWrapper.create().success(true)
	}

}
